orangecontrib/spectroscopy/utils/Peak_line.py

```diff
@@ -0,0 +1,569 @@
+# -*- coding: utf-8 -*-
+from AnyQt import QtGui, QtCore
+from pyqtgraph import Point
+from pyqtgraph import GraphicsObject
+from pyqtgraph import TextItem
+from pyqtgraph import ViewBox
+from pyqtgraph import functions as fn
+import numpy as np
+
+__all__ = ['PeakLine', 'PeakLabel']
+
+"""
+    A modification of the Infinite Line Graphicsobject.
+    Uses the base function as a means to create labels specific for peak
+    marking in spectra. 
+"""
+
+
+class PeakLine(GraphicsObject):
```

Comment:
Thanks for the PR. This file mostly seems to be a copy of the Infinite Line from pyqtgraph. This will be very hard to main. Instead, the infinite line should be subclassed, because the small modification the subclass will do are likely to be sufficient.

If subclassing is not possible, all changes to the pyqtgraph code need to clearly marked with a very visible comment.